pallets/did/src/lib.rs

```diff
@@ -256,6 +279,15 @@ pub mod pallet {
 	#[pallet::getter(fn get_did)]
 	pub type Did<T> = StorageMap<_, Blake2_128Concat, DidIdentifierOf<T>, DidDetails<T>>;
 
+	/// The set of DIDs that have been deleted and cannot therefore be created
+	/// again for security reasons.
+	///
+	/// It maps from a DID identifier to a unit tuple, for the sake of tracking
+	/// DID identifiers.
+	#[pallet::storage]
+	#[pallet::getter(fn get_deleted_did)]
+	pub(crate) type DeletedDids<T> = StorageMap<_, Blake2_128Concat, DidIdentifierOf<T>, ()>;
```

Comment:
Yeah this storage is for the system to work correctly, so it should not concern users. I am not 100% in the deposit discussion, but I think this would be stored on the chain indefinitely without a corresponding deposit, as also Shawn suggested. The deposit should be given fully back to the user when they delete their DID.